src/sensors/src/PredictSensorsMeasurements.cpp

```diff
@@ -0,0 +1,185 @@
+{
+    bool returnVal = true;
+    // incorporating gravity into the base LinAcceleration
+    iDynTree::LinkPositions linkPos(model);
+    iDynTree::LinkVelArray linkVel(model);
+    iDynTree::LinkAccArray linkAcc(model);
+    
+    iDynTree::AngAcceleration nullAngAccl;
+    iDynTree::SpatialAcc gravityAccl(gravity,nullAngAccl);
+    
+    robotAcc.baseAcc() = robotAcc.baseAcc() - gravityAccl;
+    
+    // calling ForwardPosVelAccKinematics with arguments
+    ForwardPosVelAccKinematics(model,traversal,robotPos,robotVel,robotAcc,linkPos,linkVel,linkAcc);
+    
+    // looping though SensorList    
+    unsigned int numAccl = sensorsList.getNrOfSensors(iDynTree::ACCELEROMETER);
+    unsigned int numGyro = sensorsList.getNrOfSensors(iDynTree::GYROSCOPE);
+    int idx;
+    
+    //Iterate through each kind of accelrometer and find its parent. Compute local (classical accelration) 
+    // It is automatically proper acceleration since gravity is incorporated into the base acceleration
+
+    
+    std::map<unsigned int,Sensor*> acceleroMap;
+    std::map<unsigned int,Sensor*> gyroMap;
+    Sensor * sensor;
+    for(idx = 0; idx<numAccl; idx++)
+    {
+        sensor = sensorsList.getSensor(iDynTree::ACCELEROMETER, idx);
+        acceleroMap[sensor->getParentIndex()] = sensor;
+    }
+    for(idx = 0; idx<numGyro; idx++)
+    {
+        sensor = sensorsList.getSensor(iDynTree::GYROSCOPE, idx);
+        gyroMap[sensor->getParentIndex()] = sensor;
+    }
+    
+    unsigned int vecSize = (acceleroMap.size()+ gyroMap.size()) * 3;
+    unsigned int vecItr = 0;
+    predictedMeasurement.resize(vecSize);
+    for(int i =0; i<model.getNrOfDOFs();i++)
+    {
+        // following the convention of BERDY, we iterate through the links from 
+        // root to leaves and fill in the sensor info in each (accelero first and 
+        // gyro second.
+        
+        if(acceleroMap[i] != NULL)
+        {
+            iDynTree::LinAcceleration measuredAcc, crossProdTerm;
+            iDynTree::Transform link_T_sensor;
+            iDynTree::SpatialAcc linkAccTerm;
+            iDynTree::Twist localVelocity;
+            Accelerometer *accelerometer = (Accelerometer*)acceleroMap[i];
+            returnVal = accelerometer->getLinkSensorTransform(link_T_sensor);
+            localVelocity = link_T_sensor*linkVel(i);
+            linkAccTerm = link_T_sensor*(linkAcc(i));
+            crossProdTerm = (localVelocity.getAngularVec3()).cross(localVelocity.getLinearVec3());
+            measuredAcc = (linkAccTerm).getLinearVec3() + crossProdTerm;  
+            predictedMeasurement.setVal(vecItr++,measuredAcc.getVal(0));
+            predictedMeasurement.setVal(vecItr++,measuredAcc.getVal(1));
+            predictedMeasurement.setVal(vecItr++,measuredAcc.getVal(2));
```

Comment:
setVal/getVal are the accessor with boundary checking (see http://wiki.icub.org/codyco/dox/html/idyntree/html/classiDynTree_1_1IVector.html#ae5e82c548aa91a66365888748f4b22b7), while the usual `operator()` is the one without boundary checking. I think in this case it make sense to use the accessor without boundary checking (even because they are much more readable).